### PR TITLE
Produce an error when __makeref is used and TypedReference type is no…

### LINKED ISSUE #3539: Expose CodeFix and CodeRefactoring API

### PR DESCRIPTION
**Pull request by VSadov**
…t available.

fixes #3539

### CODE REVIEW

**Comment by VSadov** [comment]
@AlekseyTs @gafter @jaredpar @agocke - please review

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenMscorlib.cs [hunk 1]
```diff
@@ -252,6 +252,44 @@ static int M(decimal d)
...
+
+        // ... method body
+
+        return result;
+    }
+}
+";
+            var compilation2 = CreateCompilation(source2, new[] { reference1 });
+            // Should report "CS0656: Missing compiler required member 'System.Decimal.op_Explicit_ToInt32'".
+            // Instead, we report no errors and assert during emit.
```
Comment (jaredpar) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenMscorlib.cs]: I don't quite understand this comment.  It says we report no errors but then just below we verify an error is produced.

Comment (VSadov) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenMscorlib.cs]: oops, I have copied this part of the test from another and the comment remained.

**Comment by gafter** [comment]
LGTM

**Comment by VSadov** [comment]
test this please

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**